Clarification of narrative, adding brief mentions to automated unit testing, and providing response to reviewer comments (#307)
SORTEE-Github-Hackathon/manuscript: content/04.use-cases-b.md

While public repositories on GitHub make it easy to store and share data files, they are not considered long-term repositories for research materials.
This is because GitHub, a for-profit company, does not have long-term data availability guarantees, allowing users to delete or make repositories private after publication.
Also, GitHub does not issue Digital Object Identifiers (DOI) for content uploaded to their servers.

> DOI are persistent and citable unique alpha-numeric identifiers assigned to digitally-stored research materials.

Reviewer: It's correct to refer to "DOIs" not "DOI"
```suggestion
DOIs are persistent and citable unique alpha-numeric identifiers assigned to digitally-stored research materials.
```

Because of this, scientists sharing code and data through GitHub are strongly encouraged to independently submit their research materials to long-term data archives (_e.g._, Zenodo, Figshare, Dryad, OSF [@doi:10.1038/538127a; @doi:10.1371/journal.pcbi.1004947; @doi:10.1029/2021EA001797]; [Table 1](#tbl:compare)).

> Some of these options (Zenodo, Figshare and OSF) integrate with GitHub, allowing project, code, and data releases ([Box 1](#definitions)) to be archived with versioned, citable DOI.

Reviewer: ```suggestion
Some of these options (Zenodo, Figshare and OSF) integrate with GitHub, allowing project, code, and data releases ([Box 1](#definitions)) to be archived with versioned, citable DOIs.
```

<!--*Contributors to this section: Kaitlyn Gaynor, Rob Crystal-Ornelas, Ali, Allison Binley*-->

GitHub can serve as a platform for everyone working with research (_e.g._, supervisors or advisors, graduate students, postdoctoral fellows, and collaborators) to share in-progress work, and flag specific challenges or questions for each other ([Table 2](#tbl:roles)).
Periodic code, data and text review is useful for identifying errors early in the research process [@doi:10.1145/3341525.3387370], and informing further training and mentorship to fill gaps in skills.
This is facilitated by a group of core features of Git and GitHub that allow contributors to discuss and simultaneously work on a project.
For instance, users can clone and fork repositories (see [Box 1](#definitions)), allowing for modifications to be done on a linked copy of a repository, which can then later be merged into the main project through pull requests.

> Collaborators can comment on specific lines of code, text being changed, suggest changes, which can then be implemented with the click of a button, greatly facilitating peer review.

Reviewer: ```suggestion
Collaborators can comment on specific lines of code or suggest changes, which can then be incorporated with the click of a button, greatly facilitating peer review.
```

Explicit project organization and increased communication within pull requests, in GitHub Issues, or in GitHub Discussions can help with project development and with potential merge conflicts due to users simultaneously working on the same sections.
Moreover, version control [@https://sortee-github-hackathon.github.io/manuscript/v/latest/index.html#storing-and-archiving-version-controlled-data] allows researchers to make changes in code or documents without worrying about irreparably modifying someone else's work.

By enabling more comprehensive remote collaboration, GitHub encourages the exchange of ideas among researchers at different institutions and in different countries, which can serve to improve the quality of the research itself by providing open access to data and code.

### Writing a manuscript{#writing-manuscript}

<!--*contributors to this section:* Emma J. Hudgins, Katherine Hébert, PHPB, Eric-->

Beyond supporting collaborative code development, GitHub can be used for writing manuscripts.
Writing a manuscript and storing its associated data and code in GitHub increases scientific reproducibility because text, code, and data can be found in one place.

> Although more time-consuming when compared to using conventional text processors, GitHub has many features that can allow for a powerful collaborative workflow when writing manuscripts [@doi:10.1186/1751-0473-8-7].

Reviewer 2: I'm not even sure that it's more time consuming once you factor in 5 collaborators all emailing you revisions on 3 different version of a Word doc.  It might be better to say something like "Using GitHub for collaborative writing may involve more initial setup and involves sacrificing some features of word processors...".  I'm ok with it as is though.

Reviewer 3: This is a good point - focusing on the initial time invested is key here I think, because once the repo is set up and the co-authors know how to contribute and review the writing, it is hard to make a case that the process becomes longer. This little nuance would avoid giving the false impression that the whole writing process becomes heavier via GitHub, which might be a little discouraging.

Maybe a bridge between these two ideas, and one that surfaces in the conclusion of the cited paper, is: 

"Although it may involve more initial time investment, GitHub has many features that support a powerful collaborative workflow when writing manuscripts [@doi:10.1186/1751-0473-8-7]"

Author: Thank you both! I have added the suggestion to the file!

Text documents stored and versioned in GitHub can be instantly displayed when written in Markdown, a lightweight markup language increasingly popular among scientists. 
When manuscripts are written in Markdown on GitHub, co-authors can contribute changes or suggest revisions to a manuscript in the same ways they would contribute changes or suggest revisions to code (see [Collaborative and asynchronous code editing](#code-editing)).
Relevant literature or issues can be suggested using the Discussions and Issues features.

Incorporating GitHub into the process of writing a manuscript does not necessarily mean pivoting to an entirely new workflow. 

> For instance, authors who prefer writing in LaTeX or Markdown can link their new or existing projects to GitHub via the Git bridge and GitHub Synchronization features ([Table 1](#tbl:compare)), which are available through premium subscriptions and university partnerships to Overleaf   [@url:https://www.overleaf.com/learn/how-to/Using_Git_and_GitHub], through HackMD (<https://hackmd.io>), and others. 

Reviewer 2: Feels like maybe too much detail.  Might be enough to mention that both Overleaf and HackMD have ways of synchronizing with GitHub
```suggestion
For instance, authors who prefer writing in LaTeX or Markdown can synchronize Overleaf,  [@url:https://www.overleaf.com/learn/how-to/Using_Git_and_GitHub], HackMD (<https://hackmd.io>), and other platforms directly with a GitHub repository.
```

Reviewer 4: this looks like a better level of detail to me

GitHub provides an open and transparent platform that can be used for either directly providing feedback on research products or addressing changes recommended by reviewers.
GitHub Issues can be used to organize and discuss reviewer suggestions and to assign them to co-authors (_e.g._, <https://github.com/SORTEE-Github-Hackathon/manuscript/issues?q=label%3A%22Reviewer+Comment%22>).
When reviewer comments are posted as separate issues, authors can comment on the issues to discuss possible changes and assign those that are going to address the issue.
Co-authors can then integrate their edits and responses to reviewers using pull requests, which can be directly linked to the issues they address.

GitHub can also assist reviewers during the peer review process.
If the code associated with a manuscript is made available at the time of submission (_e.g._, as a link to a GitHub repository within the Data Availability Statement), peer reviewers may be able to offer more comprehensive suggestions on the code and written materials, potentially recognizing errors before publication.

> rOpenSci's efforts have resulted in many well-used R packages for ecology research including `rfishbase` [@doi:10.1111/j.1095-8649.2012.03464.x] and `taxize` [@doi:10.12688/F1000RESEARCH.2-191.V2].

Reviewer 2: Either remove this sentence also or make it fit better with the previous sentence.
